### DISCUSSION TITLE
Client SDK for the File API

### DISCUSSION THREAD

**Discussion by Magnuti** [pocketbase/pocketbase#647] [Q&A] [answered]
I see that most API endpoints have client SDK examples for both JavaScript and Dart, but the Download/Fetch file endpoint in the File API does not contain such examples. Is SDK support for file fetching coming in the future?

![image](https://user-images.githubusercontent.com/43852515/192346618-0f88cd8f-c005-4f40-8ca5-9d722ea86e11.png)

At the moment in order to download a file we need the `.records.getFileUrl()` method with the record and the filename and then manually send a HTTP request to that endpoint, right? Or are there easier ways to download files?

**Comment by ganigeorgiev** [comment, marked as answer]
You could find more info how the file url is constructed in https://pocketbase.io/docs/files-handling/#file-url.

There is no method in the SDKs to download a file because I'm not sure how useful that would be. Usually the url is enough since the files are loaded as links in client-side code. Rarely there is a situation where you'll need to download the file as raw bytes locally with the client-side application code.

Could you elaborate why the plain file url doesn't work for your use case?

**Reply by Magnuti** [reply]
You are right. For example, I can probably just use the file URL as input to an image component and not the raw byte data in order to render the image.

One case where it is suitable to store the file as raw data is caching. Say I want to cache the profile picture of the logged in user so that I don't need to download it from the file URL every time the app is opened.

**Reply by ganigeorgiev** [reply]
I think It's better to leave files download to the developers to handle on their own since it is not something specific to only PocketBase.

For mobile, there are several libraries that could help with assets cache. If you are using flutter there is the [cached_network_image package](https://pub.dev/packages/cached_network_image), for android - [Coil](https://coil-kt.github.io/coil/), [Glide](https://github.com/bumptech/glide), and probably others. 

For web, most of the times assets caching is handled automatically by browsers (usually based on the `Last-Modified` header, but I plan also to add support in the future for explicit TTL and other caching header controls). Storing manually raw data client-side in the browser usually doesn't work well, especially for large files (10mb+) and could make the client tab unresponsive in case of too much runtime memory allocations.

**Reply by Magnuti** [reply]
Well that answers my question! I see now that there is typically no need for client SDK support for the File API.
